### ISSUE TITLE
Startup Applications: does not refresh until one has closed settings

### ISSUE THREAD

**Issue by ghost** [linuxmint/cinnamon#8134]
```
 * Cinnamon version (cinnamon --version) - 3.8.9 and in Mint 19.1 beta
 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...) Mint 19
 * 32 or 64 bit: 64
 ```

**Describe the bug**
Startup Applications: does not refresh - does not show a newly added startup item - until one has closed and re-opened settings; simply closing the Startup settings window does not suffice.

**To Reproduce**

1. Open Startup Applications, via Settings.
2. Add a new startup entry, e.g. Flameshot (via the Flameshot panel icon).
3. See that (i) the change is not reflected in the Startup Applications window, even (ii) when one presses its 'back button' and then re-opens it. See that the change is only reflected when one closes the main Settings window and then re-opens Startup Applications.

**Expected behavior**

Having the new startup entry appear at stage 3i would be nice, or at least stage 3ii. Or perhaps Startup Applications could have a 'refresh' option.

**Comment by icarter09** [comment]
Cinnamon 4.0.8
Linux Mint 19.1 64bit

@CottonEaster on step 2, are you creating a custom startup entry? When I add an entry from "choose application", I see that the entry has been added to the Startup Applications window.

**Comment by CottonEaster** [comment]
I've tried just now with both a custom entry and chosen application, and both methods work. Perhaps previously I failed to notice the appearance in the list of my new startup entry because sometimes a new entry shows up where you can see and sometimes it shows up where you cannot see it; the visibility depends on size of window, what letter the new startup entry starts with and where one is in the list. So I think what we have here is user error (I apologise) combined with a less than ideal interface.

**Comment by CottonEaster** [comment]
I like your drift, but 1 would be little use of the new entry is in a part of the window not visible to the user at the time of addition.

**Comment by icarter09** [comment]
Ahhh. Very true. I guess the fix for that would be to scroll to the newly added entry since they are put into alphabetical order.

**Comment by CottonEaster** [comment]
If it's feasible, scrolling to the new entry and flashing its background would be good. It might be important to fix #7697 at the same time.
